### PR TITLE
Add query tests

### PR DESCRIPTION
**Pull request by Mikearaya**
I've added queries for

* `deliveryProviders`
* `deliveryInterfaces`
* `warehousingProviders`
* `warehousingInterface`
* `paymentProviders`
* `paymentInterfaces`

**Please remember to check the checkbox on the issue if when test cases are complete** :bowing_man:

### CODE REVIEW

**Comment by Mikearaya** [comment]
got it, and on it

**Comment by Mikearaya** [comment]
all fields are tested now :checkered_flag:

File: tests/delivery-interfaces.test.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  beforeAll(async () => {
+    [, connection] = await setupDatabase();
+    graphqlFetch = await createLoggedInGraphqlFetch(ADMIN_TOKEN);
+  });
+
+  afterAll(async () => {
+    await connection.close();
+  });
+
+  describe("For logged in users", () => {
```
Comment (pozylon) [inline tests/delivery-interfaces.test.js:21]: If this only works for admin, and I guess it does, then it should say "For administrators" or at least "Users with privileges" or something

**Comment by pozylon** [comment]
```
➜  unchained-m git:(master) npm run test

> @unchainedshop/engine@0.49.0 test /Users/pozylon/Repositories/unchained-m
> jest --runInBand

 PASS  tests/plugins-datatrans.test.js
 PASS  tests/setup-basic.test.js
 PASS  tests/auth-admin.test.js
 PASS  tests/worker.test.js (6.451 s)
 PASS  tests/product-reviews.test.js
 PASS  tests/assortments.test.js
 PASS  tests/plugins-apple-iap.test.js
 PASS  tests/filters.test.js
 PASS  tests/cart-products.test.js
 PASS  tests/quotation-workflow.test.js
 PASS  tests/auth-user.test.js
 PASS  tests/subscriptions.test.js
 PASS  tests/auth-anonymous.test.js
 PASS  tests/delivery-providers.test.js
 PASS  tests/cart-checkout.test.js
 FAIL  tests/order-deliveries.test.js
  ● Order: Deliveries › Mutation.setOrderDeliveryProvider › set order delivery provider

    expect(received).toMatchObject(expected)

    - Expected  - 1
    + Received  + 1

      Object {
        "_id": "simple-order",
        "delivery": Object {
          "provider": Object {
    -       "_id": "sendmail-delivery-provider",
    +       "_id": "simple-delivery-provider",
          },
        },
      }

      44 |         },
      45 |       });
    > 46 |       expect(setOrderDeliveryProvider).toMatchObject({
         |                                        ^
      47 |         _id: SimpleOrder._id,
      48 |         delivery: {
      49 |           provider: {

      at Object.<anonymous> (tests/order-deliveries.test.js:46:40)
          at runMicrotasks (<anonymous>)

 PASS  tests/quotations.test.js
 PASS  tests/orders.test.js
 PASS  tests/cart-discounts.test.js
 PASS  tests/setup-warehousing.test.js
 PASS  tests/setup-delivery.test.js
 PASS  tests/warehousing-provider.test.js
 PASS  tests/setup-payment.test.js
 PASS  tests/payment-provider.test.js
 PASS  tests/order-management.test.js
 PASS  tests/order-payments.test.js
 PASS  tests/user-bookmarks.test.js
 PASS  tests/public-queries.test.js
 PASS  tests/translated-filter-texts.test.js
 PASS  tests/logs.test.js
 PASS  tests/translated-assortments.test.js
 PASS  tests/cart-quotations.test.js
 PASS  tests/products.test.js
 PASS  tests/warehousing-interface.test.js
 PASS  tests/delivery-interfaces.test.js
 PASS  tests/payment-interface.test.js

Summary of all failing tests
 FAIL  tests/order-deliveries.test.js
  ● Order: Deliveries › Mutation.setOrderDeliveryProvider › set order delivery provider

    expect(received).toMatchObject(expected)

    - Expected  - 1
    + Received  + 1

      Object {
        "_id": "simple-order",
        "delivery": Object {
          "provider": Object {
    -       "_id": "sendmail-delivery-provider",
    +       "_id": "simple-delivery-provider",
          },
        },
      }

      44 |         },
      45 |       });
    > 46 |       expect(setOrderDeliveryProvider).toMatchObject({
         |                                        ^
      47 |         _id: SimpleOrder._id,
      48 |         delivery: {
      49 |           provider: {

      at Object.<anonymous> (tests/order-deliveries.test.js:46:40)
          at runMicrotasks (<anonymous>)


Test Suites: 1 failed, 35 passed, 36 total
Tests:       1 failed, 8 todo, 187 passed, 196 total
Snapshots:   0 total
Time:        25.812 s
Ran all test suites.
```

**Comment by pozylon** [comment]
I get various failed tests when re-running the tests, something is racing I'm trying to spot the issue

**Comment by pozylon** [comment]
Failed on my device a few minutes ago, don't know why but it's maybe not because of your code: 

```
 FAIL  tests/setup-basic.test.js
  ● basic setup of internationalization and localization context › user defaults

    expect(received).toMatchObject(expected)

    - Expected  - 3
    + Received  + 1

      Object {
        "country": Object {
          "isoCode": "CH",
        },
    -   "language": Object {
    -     "isoCode": "de",
    -   },
    +   "language": null,
      }

      503 |       `,
      504 |     });
    > 505 |     expect(me).toMatchObject({
          |                ^
      506 |       language: {
      507 |         isoCode: 'de',
      508 |       },

      at Object.<anonymous> (tests/setup-basic.test.js:505:16)
          at runMicrotasks (<anonymous>)

  ● basic setup of internationalization and localization context › global shop context

    expect(received).toMatchObject(expected)

    - Expected  - 3
    + Received  + 1

    @@ -4,12 +4,10 @@
          "defaultCurrency": Object {
            "isoCode": "CHF",
          },
          "isoCode": "CH",
        },
    -   "language": Object {
    -     "isoCode": "de",
    -   },
    +   "language": null,
        "userRoles": Array [
          "admin",
        ],
      }

      536 |       `,
      537 |     });
    > 538 |     expect(shopInfo).toMatchObject({
          |                      ^
      539 |       _id: 'root',
      540 |       language: {
      541 |         isoCode: 'de',

      at Object.<anonymous> (tests/setup-basic.test.js:538:22)
          at runMicrotasks (<anonymous>)
```

**Comment by pozylon** [comment]
Ah I guess I know why country/language stuff failed, it's because I've added an lru-cache recently, I will push some changes to master that disables these caches in test mode, just fix the send-mail and we can merge this

**Comment by Mikearaya** [comment]
I've added the required changes, only four tests are failing currently 2 are related with language which should be fix with the new change @pozylon will introduce.
the two other failing tests are related to datatrans API and `mutation.addToCartProduct` i don't know why the calculation is not made for `total` and `taxes` but I'll investigate.

**Pull request merged**